### PR TITLE
Provide validation rule to verify if resource discovery is configured when metric discovery is used

### LINKED ISSUE #1069: Provide validation rule for resource discovery in scraper runtime config
Provide validation rule for resource collections to fail fast in scraper.

### Required Rules
- Ensure resource discovery is configured
- Ensure resource collection exists

### PR DESCRIPTION
**Pull request by tomkerkhove**
Signed-off-by: Tom Kerkhove <kerkhove.tom@gmail.com>

Closes #1069

### CODE REVIEW

File: src/Promitor.Agents.Scraper/Validation/Steps/ResourceDiscoveryValidationStep.cs [hunk 1]
```diff
@@ -4,28 +4,38 @@
...
         public string ComponentName { get; } = "Resource Discovery";
 
         public ValidationResult Run()
         {
+            var doesDeclareResourceDiscoveryGroups = DetermineIfDiscoveryGroupsAreDefined();
             if (_configuration == null)
             {
+                if (doesDeclareResourceDiscoveryGroups)
+                {
+                    return ValidationResult.Failure(ComponentName, new List<string> {"No resource discovery is configured while discovery groups are defined to scrape"});
```
Comment (adamconnelly) [inline src/Promitor.Agents.Scraper/Validation/Steps/ResourceDiscoveryValidationStep.cs:36]: Can I maybe suggest rewording this to something like:

```
Resource discovery groups are defined in your metrics configuration, but resource discovery has not been configured. Please add a resource discovery configuration.
```

I just feel like it's a bit more explicit then, and it also points people towards a solution.

Comment (tomkerkhove) [reply src/Promitor.Agents.Scraper/Validation/Steps/ResourceDiscoveryValidationStep.cs:36]: Sure, good suggestion!
